### ISSUE TITLE
`preserveSymlinks: 'auto'`

### ISSUE THREAD

**Issue by jun-sheaf** [rollup/rollup#5427]
### Feature Use Case

We use a custom build system that uses symlinks for node module resolution, but not for relative module resolution.

### Feature Proposal

Symlinks seem to be either always follow or never follow (corresponding to `preserveSymlinks: false` and `preserveSymlinks: true`). A more robust approach would be to try and resolve without following the symlink and then following the symlink if nothing is found.

**Comment by lukastaegert** [comment]
Good idea. If we do this, we should also [update `@rollup/plugin-node-resolve` accordingly](https://www.npmjs.com/package/@rollup/plugin-node-resolve#preserving-symlinks), otherwise it will likely not work for many people. PRs welcome!

**Comment by jun-sheaf** [comment]
> [update `@rollup/plugin-node-resolve` accordingly](https://www.npmjs.com/package/@rollup/plugin-node-resolve#preserving-symlinks)

@lukastaegert WDYT about using [`import-meta-resolve`](https://www.npmjs.com/package/import-meta-resolve) for the resolution algorithm rather than trying to program it ourselves?

**Comment by jun-sheaf** [comment]
We've decided to just build our own resolver that better fits our requirements: https://github.com/mathematic-inc/rollup-plugin-import-meta-resolve

Please feel free to close this issue and associated PR.
